tests/Doctrine/Tests/ORM/Functional/OneToManyMatchingCriteriaTest.php

```diff
@@ -0,0 +1,96 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional;
+
+use Doctrine\Common\Collections\Criteria;
+use Doctrine\ORM\Mapping\FetchMode;
+use Doctrine\Tests\Models\ECommerce\ECommerceCategory;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+/**
+ * Tests sorting mechanism of Selectable::matching()
```

Comment:
Please add `@group` with the id of the github issue here

Reply:
Done, added `@group 7767`